tv_apps/device-deck/test/unit/bluetooth_manager_test.js

```diff
@@ -0,0 +1,190 @@
+/* global MockMozBluetooth, MockBTAdapter, BluetoothManager, MockEventTarget,
+          MockDiscoveryHandle */
+'use strict';
+
+require('/bower_components/evt/index.js');
+require('/shared/test/unit/mocks/mock_event_target.js');
+require('/shared/test/unit/mocks/mock_navigator_moz_bluetooth_v2.js');
+require('/js/bluetooth_loader.js');
+require('/js/bluetooth_manager.js');
+
+suite('device-deck/BluetoothManager', function() {
+
+  var realMozBluetooth;
+  var bluetoothManager;
+
+  suiteSetup(function() {
+    var MockDiscoveryHandle = function() {
+      this.ondevicefound = null;
+    };
+    MockDiscoveryHandle.prototype = Object.create(MockEventTarget.prototype);
+    window.MockDiscoveryHandle = MockDiscoveryHandle;
+
+    realMozBluetooth = navigator.mozBluetooth;
+    switchReadOnlyProperty(navigator, 'mozBluetooth', MockMozBluetooth);
+  });
+
+  suiteTeardown(function() {
+    window.MockDiscoveryHandle = undefined;
+    switchReadOnlyProperty(navigator, 'mozBluetooth', realMozBluetooth);
+  });
+
+  var switchReadOnlyProperty = function (originObject, propName, targetObj) {
+    Object.defineProperty(originObject, propName, {
+      configurable: true,
+      get: function() { return targetObj; }
+    });
+  };
+
+  var setupForMockDefaultAdapter = function(context) {
+    var mockDiscoveryHandle = new MockDiscoveryHandle();
+    MockBTAdapter.state = 'enabled';
+    MockBTAdapter.startDiscovery = function() {};
+    MockBTAdapter.stopDiscovery = function() {};
+    var startDiscoveryStub =
+      context.sinon.stub(MockBTAdapter, 'startDiscovery', function() {
+        return Promise.resolve(mockDiscoveryHandle);
+      });
+
+    var stopDiscoverySpy =
+      context.sinon.stub(MockBTAdapter, 'stopDiscovery', function() {
+        return Promise.resolve();
+      });
+
+    return {
+      startDiscoveryStub: startDiscoveryStub,
+      stopDiscoverySpy: stopDiscoverySpy,
+      mockDiscoveryHandle: mockDiscoveryHandle
```

Comment:
Although these stubs are returned and saved on another data structure (namely `mockData`
I think just call `MockBTAdapter.startDiscoveryStub.calledOnce` is more straightforward..
Do you have any considerations here?